pollux/core/src/main/scala/org/hyperledger/identus/pollux/core/service/uriResolvers/DidUrlResolver.scala

```diff
@@ -0,0 +1,132 @@
+package org.hyperledger.identus.pollux.core.service.uriResolvers
+
+import io.lemonlabs.uri.{Url, UrlPath}
+import org.hyperledger.identus.pollux.vc.jwt
+import org.hyperledger.identus.pollux.vc.jwt.*
+import org.hyperledger.identus.shared.crypto.Sha256Hash
+import org.hyperledger.identus.shared.http.{GenericUriResolverError, UriResolver}
+import org.hyperledger.identus.shared.models.PrismEnvelopeData
+import org.hyperledger.identus.shared.models.StatusCode
+import zio.*
+import zio.json.*
+
+class DidUrlResolver(httpUrlResolver: HttpUrlResolver, didResolver: DidResolver) extends UriResolver {
+  import DidUrlResolver.*
+
+  def resolve(uri: String): IO[GenericUriResolverError, String] = {
+
+    for {
+      parsed <- ZIO.fromTry(Url.parseTry(uri)).mapError(_ => InvalidURI(uri))
+      maybeResourceService = parsed.query.param("resourceService")
+      maybeResourcePath = parsed.query.param("resourcePath")
+      maybeResourceHash = parsed.query.param("resourceHash")
+      serviceAndPath <- ZIO
+        .fromOption(maybeResourceService zip maybeResourcePath)
+        .mapError(_ => MissingRequiredParams(uri))
+      (resourceService, resourcePath) = serviceAndPath
+      didStr = parsed.removeQueryString().toString
+      didDocument <- didResolver.resolve(didStr).flatMap {
```

Comment:
I'm not sure this way of error detail complies with the current error-handling strategy. 
Let's commit to it as it is, but discuss it with @bvoiturier  at the AoH meeting.